src/container/container.ts

```diff
@@ -302,7 +304,7 @@ class Container implements interfaces.Container {
     }
 
     public applyMiddleware(...middlewares: interfaces.Middleware[]): void {
-        const initial: interfaces.Next = (this._middleware) ? this._middleware : this._planAndResolve();
+        const initial: interfaces.Next = (this._middleware) ? this._middleware : (args) => [ExecutePlanSymbol, args];
```

Comment:
This is an idea in progress but the end of the chain needs to return the resolved value as middleware should be able to change the resolved value.

```

        it.only("Should allow middleware to change the resolved value", () => {
            const middleware: interfaces.Middleware = (next => {
                return (nextArgs => {
                    const resolved = next(nextArgs) as string;
                    return `${resolved}+`;
                })
            })

            const container = new Container();
            container.applyMiddleware(middleware);
            container.bind("Sid").toConstantValue("One");
            expect(container.get<string>("Sid")).to.equal("One+");
        })
```

Reply:
Makes sense. I'll add this test.